raiden/network/rpc/client.py

```diff
@@ -797,3 +797,15 @@ def check_for_insufficient_eth(
             msg = f'Failed to execute {transaction_name} due to insufficient ETH'
             log.critical(msg, required_wei=required_balance, actual_wei=balance)
             raise InsufficientFunds(msg)
+
+    def get_checking_block(self):
+        """Workaround for parity https://github.com/paritytech/parity-ethereum/issues/9707
+        In parity doing any call() with the 'pending' block no longer falls back
```

Comment:
why do we want to do calls with the `pending` block anyways?

Reply:
To see the value of the blockchain state (i.e. a token balance e.t.c.) in the pending block.

Reply:
Yeah, I understand how it works, I don't see why do we want to use it like that.

Reply:
> It's usefull when you want to check the attributes of an onchain state in order to make a transaction based on that state

That doesn't make much sense to me, the pending state is not mined yet, the Ethereum node Raiden is talking probably won't mine the next block, so even if the call is (in)valid during the `eth_call` there is no guarantee it will be that way when the transaction is mined.

> As a silly example, to check if a channel is opened because you want to make a deposit you would check on the pending block because your deposit transaction would always go on the pending block and not on the latest (which is a block that has already been mined)

That is what the `eth_estimateGas` call is for, isn't it? If the channel is not open, as far I understand from our previous discussions, the gas estimation will be `None`

Reply:
> That doesn't make much sense to me, the pending state is not mined yet, the Ethereum node Raiden is talking probably won't mine the next block, so even if the call is (in)valid during the eth_call there is no guarantee it will be that way when the transaction is mined.

Sure. But the pending block will always be closer to the transaction's mined block than the latest block. The transaction can never be mined in the latest block. Dapps generally tend to check stuff on the `pending` block.

> That is what the eth_estimateGas call is for, isn't it? If the channel is not open, as far I understand from our previous discussions, the gas estimation will be None

Yes that is correct. With estimate gas the blockchain precondition checks don't make much sense since estimate gas will give us a yes/no answer anyway.

Reply:
> But the pending block will always be closer to the transaction's mined block than the latest block

I don't follow, what do you mean by closer?

> Dapps generally tend to check stuff on the pending block.

Okay, but why? What is the benefit here?

Reply:
> I don't follow, what do you mean by closer?

Say our local ethereum client is now looking at block `L` as the `latest` mined block it knows about. We are about to send a transaction. This transaction will either get mined in the `pending` block, so `L+1` or anything after that. The pending block is closer in distance of blocks to the block the transaction we want to send will eventually get mined in.

So as far as state is concerned that block will have a closer to reality state as opposed to the latest block. 

> Okay, but why? What is the benefit here?

> So as far as state is concerned that block will have a closer to reality state as opposed to the latest block.